app_dart/lib/src/service/cache_service.dart

```diff
@@ -56,7 +59,9 @@ class CacheService {
     final Cache<Uint8List> subcache = cache.withPrefix(subcacheName);
     Uint8List? value;
 
+    await m.acquireRead();
```

Comment:
Yeah based on your other comment this might be a good thing to do!

Reply:
After thinking about this some more I think it would be better to create a different class that specifically has locking in it that the calls for tokens will use. I think this is the safer method.

Reply:
Hmm okay after looking through the code I will go ahead with the locking methods.